File: verification/verify_ui_restoration.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        print("Navigating to Home...")
        page.goto("http://localhost:4321/")

        # Verify Hero Headline
        if page.locator(".hero-headline h1").count() > 0:
            print("✅ Hero Headline found")
        else:
            print("❌ Hero Headline MISSING")

        # Verify Brand Animation
        if page.locator(".brand-tetris-container").count() > 0:
            print("✅ Brand Animation found")
        else:
            print("❌ Brand Animation MISSING")

        # Verify Quick Zones
        zones = page.locator(".zone-chip").count()
        if zones >= 3:
            print(f"✅ Quick Zones found: {zones}")
        else:
            print(f"❌ Quick Zones missing or too few: {zones}")

        # Verify Community Link
        if page.locator('a[href="/community"]').count() > 0:
            print("✅ Community Link found")
        else:
            print("❌ Community Link MISSING")

        print("Navigating to Catalog...")
        page.goto("http://localhost:4321/rutas")

        # Verify Routes
        routes = page.locator(".glass-card").count()
        if routes > 2:
            print(f"✅ Routes loaded correctly: {routes}")
        else:
            print(f"❌ Routes missing (only {routes} found)")

        # Screenshot
        page.screenshot(path="verification/ui_restored.png", full_page=True)
        print("📸 Screenshot saved to verification/ui_restored.png")

        browser.close()

if __name__ == "__main__":
    run()

File: verification/verify_ui_restoration.mjs
import { test, expect } from '@playwright/test';
import path from 'path';

test('Verify UI Restoration', async ({ page }) => {
  // 1. Go to Home
  await page.goto('http://localhost:4321/');

  // 2. Check Hero Headline
  const hero = page.locator('.hero-headline h1');
  await expect(hero).toContainText('¿Qué ruta me lleva?');
  await expect(hero).toContainText('Cancún');

  // 3. Check Brand Animation (Top Bar Mobile or Desktop Header)
  // Depending on viewport, either .brand-tetris or header .brand-tetris-container exists
  const brand = page.locator('.brand-tetris, .brand-tetris-container').first();
  await expect(brand).toBeVisible();

  // 4. Check Quick Access Zones in Search
  const quickZones = page.locator('.zone-chip');
  await expect(quickZones).toHaveCount(4); // Cancún, Pto Morelos, Riviera Maya, Aeropuerto

  // 5. Check Navigation (Community Link)
  const navLink = page.locator('a[href="/community"]');
  await expect(navLink).toBeVisible();

  // 6. Go to Catalog (Rutas)
  await page.goto('http://localhost:4321/rutas');

  // 7. Verify Routes Loaded (More than 2)
  const routeCards = page.locator('.glass-card');
  const count = await routeCards.count();
  console.log(`Found ${count} route cards`);
  expect(count).toBeGreaterThan(2);

  // 8. Go to Community
  await page.goto('http://localhost:4321/community');
  await expect(page.locator('h1')).toContainText('Comunidad');

  // Screenshot
  await page.screenshot({ path: 'verification/ui_restored.png', fullPage: true });
});
